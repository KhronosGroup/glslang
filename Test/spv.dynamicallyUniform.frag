#version 450

#extension GL_EXT_dynamically_uniform_attribute : enable

[[dynamically_uniform]] int foo([[dynamically_uniform]] int x) {
  return x;
}

void main() {
  {
    int x = 42;
    int y;
    int z[4];
    int w[1] = {42};
  }

  {
    [[dynamically_uniform]] int x = 42;
    [[dynamically_uniform]] int y;
    [[dynamically_uniform]] int z[4];
    [[dynamically_uniform]] int w[1] = {42};

    y = 13;
    z[0] = foo(y);
  }
}
