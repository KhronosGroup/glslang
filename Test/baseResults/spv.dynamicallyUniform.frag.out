spv.dynamicallyUniform.frag
// Module Version 10000
// Generated by (magic number): 80006
// Id's are bound by 36

                              Capability Shader
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Fragment 4  "main"
                              ExecutionMode 4 OriginUpperLeft
                              Source GLSL 450
                              SourceExtension  "GL_EXT_dynamically_uniform_attribute"
                              Name 4  "main"
                              Name 10  "foo(i1;"
                              Name 9  "x"
                              Name 15  "x"
                              Name 21  "w"
                              Name 23  "x"
                              Name 24  "w"
                              Name 25  "y"
                              Name 30  "z"
                              Name 32  "param"
                              Decorate 9(x) Uniform
                              Decorate 12 Uniform
                              Decorate 22 Uniform
                              Decorate 23(x) Uniform
                              Decorate 24(w) Uniform
                              Decorate 22 Uniform
                              Decorate 25(y) Uniform
                              Decorate 30(z) Uniform
                              Decorate 33 Uniform
                              Decorate 34 Uniform
                              Decorate 34 Uniform
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeInt 32 1
               7:             TypePointer Function 6(int)
               8:             TypeFunction 6(int) 7(ptr)
              16:      6(int) Constant 42
              17:             TypeInt 32 0
              18:     17(int) Constant 1
              19:             TypeArray 6(int) 18
              20:             TypePointer Function 19
              22:          19 ConstantComposite 16
              26:      6(int) Constant 13
              27:     17(int) Constant 4
              28:             TypeArray 6(int) 27
              29:             TypePointer Function 28
              31:      6(int) Constant 0
         4(main):           2 Function None 3
               5:             Label
           15(x):      7(ptr) Variable Function
           21(w):     20(ptr) Variable Function
           23(x):      7(ptr) Variable Function
           24(w):     20(ptr) Variable Function
           25(y):      7(ptr) Variable Function
           30(z):     29(ptr) Variable Function
       32(param):      7(ptr) Variable Function
                              Store 15(x) 16
                              Store 21(w) 22
                              Store 23(x) 16
                              Store 24(w) 22
                              Store 25(y) 26
              33:      6(int) Load 25(y)
                              Store 32(param) 33
              34:      6(int) FunctionCall 10(foo(i1;) 32(param)
              35:      7(ptr) AccessChain 30(z) 31
                              Store 35 34
                              Return
                              FunctionEnd
     10(foo(i1;):      6(int) Function None 8
            9(x):      7(ptr) FunctionParameter
              11:             Label
              12:      6(int) Load 9(x)
                              ReturnValue 12
                              FunctionEnd
